### PR TITLE
[WIP] Uplink Redone! - Remake of Pull - #9197

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
#9197

## About The Pull Request

Wew - Makes each topic in the uplink into their own folder with their own way to sort it
Gives more ammo 12g/boolets ect ect to Shotgun, SMGs. .357 and LAW6
Gives items to bundles + more bundles 
Changes the name of some ammo types
Changes SMG C-40 to 9mms from 45s
Rebrands a few things
More uplink things
Makes snipers have 10 more harm but 1 less boolet
Makes a new VP-2 shot 9mm gun, puts away the ASP
Brakes a lot of icons maybe

## Why It's Good For The Game

Makes uplink look nicer, has more things in it that done take divivng into things 
changing things ect ect more at #9197 

## Changelog
:cl:
add: Added many new things
add: Added more things
tweak: harm of ammo and ammo amounts 
balance: many things
spellcheck: a few things ya
code: more files
refactor: maybe more files?
/:cl:

### CODE REVIEW

**Comment by Poojawa** [comment]
Don't make meme changelogs

**Comment by Trilbyspaceclone** [comment]
@Poojawa Not a meme one

**Comment by Ghommie** [comment]
Poojawa meant the changelog should be concrete and actually inform players (and to a lesser extend admins and then contributors) about the changes. While the autochangelog hasn't been working for a long while (the .yml files are being made but the server doesn't want to compile them), it's still good practice to write down a moderately detailed explaination of the proposed changes as not every player is willing to code dive those infos.

By the by, as I probably told Tupinambis before he closed his own PR, albeit locating different categories into their own specific files can be considered an index improvement, it modifies the position of each category displayed on the UI as they'll be loaded in a different order now. Plus the new additions and blundles currently do not follow the pseudo-alphabetical indexing that has been previously introduced to uplink items.

TL;DR:
Move the category datums back to uplink_items.dm in their former position, and enforce the alphabetic order on the new additions where convenient. Also less vague changelog, please.

**Comment by Trilbyspaceclone** [comment]
@Ghommie Ye working on the sprites first, and if you take a look its more based on "Whom can use it before ABC so that clown ops gear / Nukies show on top rather then digging in.

**Comment by Ghommie** [comment]
> 
> 
> @Ghommie Ye working on the sprites first, and if you take a look its more based on "Whom can use it before ABC so that clown ops gear / Nukies show on top rather then digging in.

An alphabetic order is easier to follow than a "whose is this item targetted to." one, for items within a category, no need to get overly finnicky.
Trying to follow both methods will only result in a harder to follow cluster for both player and contributor.
I'm not upset much about the categories being reclassified alphabetically, but their contents' order being scrambled around, which makes it harder to search for a specific item when combined with the reclassified categories without ctrl f.

Overall, go ahead. I conclude we'll have to automate the index with TimSort at some point anyway.

**Comment by Trilbyspaceclone** [comment]
@Ghommie Uplink order is fixed and now its in ABC + Whom can use it
This makes clownops/nukies much easyer to see what is their items vs whats normal

**Comment by YakumoChen** [comment]
Would still prefer you not change the ammo type names. TG code doesn't conform to hyper realism and it will only serve to confuse the hell out of everyone.

As was also stated, you should atomize the PR, throwing a bunch of balance and features into a big pr is going to make it rot and isn't good practice.

**Comment by Ghommie** [comment]
Made exception for the LMG ammo calibers (the former caliber was actually absurd unless the gun was specifically meant to shoot pencils), I can agree with that, Yakumo.

**Comment by Trilbyspaceclone** [comment]
Ammo caliber dosnt matter other then the C-40 being weakend to 10mms also
@YakumoChen Atomic prs are worst then normal as stated by Kevin whom stated as much so bigger prs will be handed slower but better

**Comment by Trilbyspaceclone** [comment]
Anymore conserns with balance and or naming, please ping me on discord.
Any more coding problems please leave a review

**Comment by YakumoChen** [comment]
@Trilbyspaceclone it does matter, as I said, changing all the calibers will confuse the playerbase and not do much else. We aren't going for realism. If caliber doesn't matter, why are you changing it at all?
